lighthouse-core/audits/errors-in-console.js

```diff
@@ -42,11 +44,43 @@ class ErrorLogs extends Audit {
     };
   }
 
+  /** @return {AuditOptions} */
+  static defaultOptions() {
+    return {
+      ignoredPatterns: undefined,
+    };
+  }
+
+
+  /**
+   * @template {{description: string | undefined}} T
+   * @param {Array<T>} items
```

Comment:
this is only ever called with `/** @type {Array<{source: string, description: string|undefined, url: string|undefined}>} */`, which is already declared once. Should just be a typedef instead of parametric? :)

Reply:
why though? this method operates on things that have an optional description, no matter what they look like, and that's exactly how the types are expressed

Reply:
> why though? this method operates on things that have an optional description, no matter what they look like, and that's exactly how the types are expressed

this is a nit, so can leave it, but it's only generic in the sense that all monomorphic functions are generic (over one type). Adding a parameter implies complexity that isn't there (it's only ever called with one type of thing).

Reply:
true, and if typescript had another method of expressing monomorphic functions that operate on a subset of the type while preserving the output type I would happily switch to that method :)

Reply:
> :)

![image](https://media1.tenor.com/images/82fcb4b08428d890e2c7c492cd53c923/tenor.gif?itemid=5307810)

Reply:
> method of expressing monomorphic functions that operate on a subset of the type while preserving the output type

I mean...
```js
/**
 * @param {Array<ConcreteType>} items
 * @param {AuditOptions} options
 * @return {Array<ConcreteType>}
 */
static filterAccordingToOptions(items, options) {
  // ...
}
```
operates on a subset of the type just fine :P

(I blame everyone's emojis for the continuation of this conversation 👻📠)